Support for arm32v7

christian-eriksson opened the pull request:
Add a new option `-a` to the build script to allow the user to compile for arm32v7 architecture as well as amd64. If `-a` flag is omitted amd64 is still the default.

To test I've build with the `-a arm32v7` on my Ubuntu 18.04 laptop and successfully installed and am running the result on my raspberry pi 4, running a 32-bit version of DietPi (Debian).

I saw there was an open issue (#12) for armhf support and a fork that seemed to work to I thought I'd create a PR, inspired by that fork. Have a look and see if it might be something for this project.

christian-eriksson committed:
Move Dockerfile.patch files to architecture dependent directories

Need a way to distinguish between patches when we implement support for other architectures.
Putting them in different paths seems simple and straight forward enough.

christian-eriksson committed:
Add -a option to support arm32v7 and amd64 architectures

Previous releases lacked support for compiling for arm32v7 architecture.
In order to keep previous behaviour the default is set to amd64. Using the
-a option, users can now choose between arm32v7 and amd64. If arm32v7 is
used only the Buster release of debian is supported. The limitation is
due to the libmariadb3:armhf dependency not being avaliable in earlier
debian releases.

christian-eriksson committed:
Update documentation to detail the -a option

christian-eriksson committed:
Remove reference to local dirs

christian-eriksson committed:
Set debian package to report correct architecture for arm32v7 build

christian-eriksson committed:
Format arm32v7 Dockerfile.patch to match the amd64

greizgh commented:
Thank you for contributing :+1:

greizgh commented on debian/control-armhf:
```diff
@@ -0,0 +1,7 @@
+Architecture: armhf
+Maintainer: Greizgh <greizgh+bitwardenrs@ephax.org>
+Priority: optional
+Version: 1.17.0-1
```
This should not be hardcoded, see [here](https://github.com/greizgh/bitwarden_rs-debian/blob/master/build.sh#L54) how the version is handled.
We should have something similar for the architecture.

christian-eriksson replied:
Ah, good catch. I've removed the armhf specific control file and am letting the build script update it as with the version property instead.

christian-eriksson committed:
Handle architecture property in control file dynamically

Similar to version, the architecture property in the control file should be
handled dynamically. The build script will now replace the architecture prop
from control.dist to match the choosen architecture.

greizgh commented:
Thank you @christian-eriksson
